src/meta/src/rpc/service/ddl_service.rs

```diff
@@ -233,6 +237,156 @@ where
+            let mut dependent_relations = Default::default();
+            for fragment in fragment_graph.fragments.values() {
+                resolve_dependent_relations(
+                    fragment.node.as_ref().unwrap(),
+                    &mut dependent_relations,
+                )
+                .map_err(tonic_err)?;
+            }
+            assert!(
+                !dependent_relations.is_empty(),
+                "there should be at lease 1 dependent relation when creating materialized view"
+            );
+            mview.dependent_relations = dependent_relations.into_iter().collect();
+        }
+
+        // 2. Mark current mview as "creating" and add reference count to dependent relations.
+        // self.catalog_manager
+        //     .start_create_table_procedure(&mview)
+        //     .await
+        //     .map_err(tonic_err)?;
+
+        // 3. Create mview in stream manager. The id in stream node will be filled.
+        let mut ctx = CreateMaterializedViewContext {
+            schema_id: mview.schema_id,
+            database_id: mview.database_id,
+            mview_name: mview.name.clone(),
+            table_properties: mview.properties.clone(),
+            affiliated_source: None,
+            ..Default::default()
+        };
+        let internal_tables = match self
+            .create_sink_on_compute_node(fragment_graph, id, &mut ctx)
+            .await
+        {
+            Err(e) => {
+                // self.catalog_manager
+                //     .cancel_create_table_procedure(&mview)
+                //     .await
+                //     .map_err(tonic_err)?;
+                return Err(e.into());
+            }
+            Ok(()) => {
+                // self.set_table_mapping(&mut mview).map_err(tonic_err)?;
+                let mut internal_table = ctx
+                    .internal_table_id_map
+                    .iter()
+                    .filter(|(_, table)| table.is_some())
+                    .map(|(_, table)| table.clone().unwrap())
+                    .collect_vec();
+
+                for inner_table in &mut internal_table {
+                    self.set_table_mapping(inner_table).map_err(tonic_err)?;
+                }
+                internal_table
+            }
+        };
+
+        // tracing for checking the diff of catalog::Table and internal_table_id count
+        tracing::info!(
+            "create_materialized_view internal_table_count {} internal_table_id_count {}",
+            internal_tables.len(),
+            ctx.internal_table_id_map.len()
+        );
+
+        // // 4. Finally, update the catalog.
+        // let version = self
+        //     .catalog_manager
+        //     .finish_create_table_procedure(internal_tables, &mview)
+        //     .await
+        //     .map_err(tonic_err)?;
+
+        // Ok(Response::new(CreateMaterializedViewResponse {
+        //     status: None,
+        //     table_id: id,
+        //     version,
+        // }))
+
```

Comment:
Clean-up unused code?

Reply:
I guess this was leftover code from copying catalog implementation in `create_materialized_view`. I've replaced it with the corresponding catalog functionality for Sink.